dspace-api/src/test/java/org/dspace/discovery/DiscoveryIT.java

```diff
@@ -273,6 +281,35 @@ public void solrRecordAfterDeleteTest() throws Exception {
 
     }
 
+    @Test
+    public void solrRecordFromMessyItemTest() throws Exception {
+        configurationService.setProperty("authority.controlled.dc.subject", "true");
+        metadataAuthorityService.reset();
+        try {
+            context.turnOffAuthorisationSystem();
+
+            parentCommunity = CommunityBuilder.createCommunity(context)
+                                              .withName("Parent Community").build();
+
+            Collection col1 = CollectionBuilder.createCollection(context, parentCommunity)
+                                               .withName("Collection 1").build();
+
```

Comment:
Could we provide a starting state assertion here? I'd like to see `assertSearchQuery(IndexableItem.TYPE, 0);` being added here so that we can prove that we didn't have any Items present in SOLR before creating the one with the "non-existing" authority value.
This way, we're always sure that the item being indexed is the item within the test, since if it wasn't then the starting state assertion would have failed. This protects us from the use case where the actual indexing of this authority value would break, but the test could succeed because an Item in a previous test wasn't getting cleared correctly.